src/transport/message_handler.rs

```diff
@@ -0,0 +1,210 @@
+use crate::events::{Event, EventHandler};
+use crate::transport::msgs::{LSPSMessage, Prefix, RawLSPSMessage, LSPS_MESSAGE_TYPE};
+use bitcoin::secp256k1::PublicKey;
+use lightning::ln::peer_handler::CustomMessageHandler;
+use lightning::ln::wire::CustomMessageReader;
+use lightning::log_info;
+use lightning::util::logger::Logger;
+use std::collections::HashMap;
+use std::convert::{TryFrom, TryInto};
+use std::io;
+use std::ops::Deref;
+use std::sync::{Arc, Mutex};
+
+/// A trait used to implement a specific LSPS protocol
```

Comment:
Mh, I think this needs much better docs: how does this relate to `TransportMessageHandler` exactly? Since `ProtocolMessageHandler` is a general trait that *each* protocol needs to implement, should this rather live in the general namespace, for example directly in`lib.rs` for the time being?

nit: Let's keep the first descriptive line separate, which improves display in module-level docs.
```suggestion
/// A trait used to implement a specific LSPS protocol.
///
```